File: etc/~2020.11.26 - codingsample/202008/20200824/Rains_of_Reason_3_variableName.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h> 

bool variableName(char * name) {
	int i=0,len = strlen(name);
	// 변수 초기화
	 
	for(i=0;i<len;i++){		
		if((isalnum(name[i]) || name[i]=='_') == false || isdigit(name[0])) return false;
	}
	return true;
}



int main() {
	char* input = "var_1__Int";
	if(variableName(input)) printf("True\n");
	else printf("False;");
}

File: etc/~2020.11.26 - codingsample/202008/20200824/Rains_of_Reason_3_variableName.py
def variableName(name):
    return name.isidentifier()

def variableName2(name):
    if name[0].isdigit() :return False
    for i in range(len(name)):
        if (name[i].isalnum() or name[i] == '_')==False: 
            return False
    return True

a = "vqqsd_5fq"
print(variableName2(a))
